### ISSUE TITLE
AddTransform requires an explicit cast to sitk.Transform

### ISSUE THREAD

**Issue by kevin-keraudren** [SimpleITK/SimpleITK#44]
Here is an illustration of the problem in ipython:

``` python

In [1]: import SimpleITK as sitk
In [2]: a = sitk.Euler3DTransform()
In [3]: b = sitk.Euler3DTransform() # or any other transform
In [4]: a.AddTransform(b)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-16-e01092ec07d0> in <module>()
----> 1 a.AddTransform(b)

[...]/envs/py35/lib/python3.5/site-packages/SimpleITK/SimpleITK.py in AddTransform(self, t)
   4882
   4883         """
-> 4884         return _SimpleITK.Transform_AddTransform(self, t)
   4885
   4886

RuntimeError: Exception thrown in SimpleITK Transform_AddTransform: /nfs/Users/blowekamp/anaconda/conda-bld/work/Code/Common/src/sitkEuler3DTransform.cxx:174:
sitk::ERROR: Transform is not of type Euler3DTransform!
```

Weird thing, if you run the last line again:

``` python

In [5]: a.AddTransform(b)
Out[5]: <SimpleITK.SimpleITK.Transform; proxy of <Swig Object of type 'itk::simple::Transform *' at 0x1033af900> >
```

A way to make things work with the current code is to write:

```
c = sitk.Transform(a).AddTransform(sitk.Transform(b))
```

A possible fix is to cast all transforms to `sitk.Transform` before adding them. The current code:

``` python

    def AddTransform(self, t):
        """
        AddTransform(Transform self, Transform t) -> Transform

        """
        return _SimpleITK.Transform_AddTransform(self, t)
```

could be replaced with:

``` python

    def AddTransform(self, t):
        """
        AddTransform(Transform self, Transform t) -> Transform

        """
        return _SimpleITK.Transform_AddTransform(Transform(self), Transform(t))
```

Although it is likely that the problem occurs with the other wrapping languages too and that a proper fix should take place in the C++ code.

**Comment by blowekamp** [comment]
This method should not be available there.

For all the classes derived Transform, in C++ the AddTransform method is made private [1]. While this is not really a "clean" C++ interface it does clearly show that the method is not intended to be used, in this way.

You are correct that the the transform should be cast/converted to the base class first to use it at a composite transform.

I add a patch to explicitly tell SWIG to ignore these methods.

These Composite transform methods are a little rough. They have been implemented so that they can easily work with the ImageRegistrationMethod usage. Additional, contribution for examples, documentation and requested methods needed to additional use cases are welcome!

[1] https://github.com/SimpleITK/SimpleITK/blob/daba3c30e4cd147565f5f8662a145ec8287e0f5d/Code/Common/include/sitkEuler3DTransform.h#L89

**Comment by blowekamp** [comment]
The composite transform functionality has been refactored into a new `CompositeTransform` class. Closed by #1124.
